File: snippets/swift/BreezSDKExamples/Sources/LnurlPay.swift
//
//  LnurlPay.swift
//
//
//  Created by ruben on 14/11/2023.
//

import BreezSDK
import Foundation

func pay(sdk: BlockingBreezServices) -> LnUrlPayResult? {
    // ANCHOR: lnurl-pay
    // Endpoint can also be of the form:
    // lnurlp://domain.com/lnurl-pay?key=val
    // lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttsv9un7um9wdekjmmw84jxywf5x43rvv35xgmr2enrxanr2cfcvsmnwe3jxcukvde48qukgdec89snwde3vfjxvepjxpjnjvtpxd3kvdnxx5crxwpjvyunsephsz36jf
    var response: LnUrlPayResult?
    let lnurlPayUrl = "lightning@address.com"
    if let inputType = try? parseInput(s: lnurlPayUrl) {
        if case .lnUrlPay(let data, let _bip353Address) = inputType {
            let amountMsat = data.minSendable
            let useTrampoline = true
            let optionalComment = "<comment>"
            let optionalPaymentLabel = "<label>"
            let req = LnUrlPayRequest(data: data, amountMsat: amountMsat, useTrampoline: useTrampoline, comment: optionalComment, paymentLabel: optionalPaymentLabel)
            response = try? sdk.payLnurl(req: req)
        }
    }
    // ANCHOR_END: lnurl-pay
    return response
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlPay.kt
package com.example.kotlinmpplib

import breez_sdk.*
class LnurlPay {
    fun lnurl_pay(sdk: BlockingBreezServices) {
        // ANCHOR: lnurl-pay
        // Endpoint can also be of the form:
        // lnurlp://domain.com/lnurl-pay?key=val
        // lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttsv9un7um9wdekjmmw84jxywf5x43rvv35xgmr2enrxanr2cfcvsmnwe3jxcukvde48qukgdec89snwde3vfjxvepjxpjnjvtpxd3kvdnxx5crxwpjvyunsephsz36jf
        val lnurlPayUrl = "lightning@address.com";
        try {
            val inputType = parseInput(lnurlPayUrl)
            if (inputType is InputType.LnUrlPay) {
                val requestData = inputType.data
                val amountMsat = requestData.minSendable
                val optionalComment = "<comment>";
                val optionalPaymentLabel = "<label>";
                val req = LnUrlPayRequest(requestData, amountMsat, optionalComment, optionalPaymentLabel)
                sdk.payLnurl(req)
            }
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: lnurl-pay
    }
}